tests/SerdeTestCases.php

```diff
@@ -1622,6 +1623,28 @@ public function nullable_properties_flattened_validate(mixed $serialized): void
 
     }
 
+    #[Test]
+    public function nullable_value_objects_flattened(): void
+    {
+        $s = new SerdeCommon(formatters: $this->formatters);
+
+        $data = new FlattenedValueObjectMain(null);
+
+        // Serialisation here hides the issue...
+        // $serialized = $s->serialize($data, $this->format);
+
+        $serialized = match ($this->format) {
+            'array' => ['order_id' => null],
+            'json' => '{"order_id":null}',
+            'yaml' => 'order_id: null',
+        };
```

Comment:
So, just to clarify how the tests are designed to work, don't do this. 😄  Instead, serialize it as the other tests do, but include a call to a _validate method, as the other tests do.  Have an empty one in the SerdeTestCases class, and then in ArrayBasedFormatterTests, fill one in that casts the results back to an array (see the others in that class) and validates that it's the right data.

I don't have time to dig into what's going on here at the moment, but having a test structure that matches the others will make it easier to investigate when I have the chance.  Thanks.